File: non_constructible_change/non_constructible_change.java
import java.util.Arrays;

public class Main {

    public static int nonConstructibleChange(int[] coins) {
        Arrays.sort(coins);
        int currentChangeCreated = 0;

        for (int coin : coins) {
            if (coin > currentChangeCreated + 1) {
                return currentChangeCreated + 1;
            }
            currentChangeCreated += coin;
        }

        return currentChangeCreated + 1;
    }

    public static void main(String[] args) {
        int[] coins = {5, 7, 1, 1, 2, 3, 22};
        System.out.println(nonConstructibleChange(coins));  
    }
}

File: non_constructible_change/non_constructible_change.go
package main

import (
	"fmt"
	"sort"
)

func nonConstructibleChange(coins []int) int {
	sort.Ints(coins)
	currentChangeCreated := 0

	for _, coin := range coins {
		if coin > currentChangeCreated+1 {
			return currentChangeCreated + 1
		}
		currentChangeCreated += coin
	}

	return currentChangeCreated + 1
}

func main() {
	coins := []int{5, 7, 1, 1, 2, 3, 22}
	fmt.Println(nonConstructibleChange(coins))  
}
